embulk-core/src/main/java/org/embulk/spi/PageBuilder.java

```diff
@@ -170,49 +146,76 @@ public void setTimestamp(int columnIndex, Timestamp value)
     {
         if (value == null) {
             setNull(columnIndex);
-            return;
         }
+        else {
+            row.setTimestamp(columnIndex, value);
+        }
+    }
 
-        int offset = getOffset(columnIndex);
-        bufferSlice.setLong(offset, value.getEpochSecond());
-        bufferSlice.setInt(offset + 8, value.getNano());
+    void writeNull(int columnIndex)
```

Comment:
Agree. I changed those modifiers with `private` since `Row` class was moved as static inner class of `PageBuilder`. d0aa12f58af67940cea6d4b39f47568473a178c6